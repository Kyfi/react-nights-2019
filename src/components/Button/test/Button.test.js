import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'

import { Button } from '../index'

// This is just an example how to test components
describe('[components] Button', () => {
  it('should render correctly', () => {
    const renderer = render(<Button>My Button</Button>)
    expect(renderer.container).toMatchSnapshot()
  })

  describe('when default', () => {
    it('should render correctly', () => {
      const renderer = render(<Button default>My Button</Button>)
      expect(renderer.container).toMatchSnapshot()
    })
  })
})
