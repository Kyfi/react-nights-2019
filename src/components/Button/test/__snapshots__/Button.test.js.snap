// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components] Button should render correctly 1`] = `
.c0 {
  background: #ef0d33;
  padding: 1rem;
  margin-top: 0.5rem;
  border: 1px solid #ef0d33;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background: transparent;
  color: #ef0d33;
}

<div>
  <button
    class="c0"
    type="button"
  >
    My Button
  </button>
</div>
`;

exports[`[components] Button when default should render correctly 1`] = `
.c0 {
  background: #ef0d33;
  padding: 1rem;
  margin-top: 0.5rem;
  border: 1px solid #ef0d33;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  background: transparent;
  color: #607d8b;
  border-color: transparent;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background: transparent;
  color: #ef0d33;
}

.c0:hover {
  background: #607d8b;
  color: #fff;
}

<div>
  <button
    class="c0"
    default=""
    type="button"
  >
    My Button
  </button>
</div>
`;
